pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -379,32 +387,10 @@ final class Event {
     String? isEmbedded,
     String? ideLaunchedFeature,
     String? isWasm,
-
-    // PerformanceScreenMetrics
-    int? uiDurationMicros,
-    int? rasterDurationMicros,
-    int? shaderCompilationDurationMicros,
-    int? traceEventCount,
-
-    // ProfilerScreenMetrics
-    int? cpuSampleCount,
-    int? cpuStackDepth,
-
-    // MemoryScreenMetrics
-    int? heapDiffObjectsBefore,
-    int? heapDiffObjectsAfter,
-    int? heapObjectsTotal,
-
-    // InspectorScreenMetrics
-    int? rootSetCount,
-    int? rowCount,
-    int? inspectorTreeControllerId,
-
-    // DeepLinkScreenMetrics
-    String? androidAppId,
-    String? iosBundleId,
+    Map<String, Object?> additionalMetrics = const {},
```

Comment:
I wonder if it would make sense to create an `abstract base class CustomMetrics` that's exposed by this package that provides a `toMap()` method (and maybe some validation logic?) and use that as the type here? This would at least provide some type safety and hopefully reduce the risk of us accidentally passing an object that isn't JSON serializable to the event.